runtime/src/test/groovy/io/micronaut/runtime/context/scope/RefreshScopeSpec.groovy

```diff
@@ -138,13 +140,15 @@ class RefreshScopeSpec extends Specification {
 
         when:
         System.setProperty("foo.bar", "bar")
+        System.setProperty("foos.test.bar", "bar")
         Environment environment = beanContext.getEnvironment()
         Map<String, Object> previousValues = environment.refreshAndDiff()
         beanContext.publishEvent(new RefreshEvent(previousValues))
 
         then:
         bean.testValue() == 'bar'
         bean.testConfigProps() == 'bar'
+        bean.testEachProps() == 'bar'
```

Comment:
Will this work for `@EachBean(EachConfig) class EachConfigBeans` ?

Reply:
And `@EachProperty(list=`?
The other use-cases might be combinations with `@Requires`

Reply:
it is unrelated, they don't config

Reply:
I don't think we can support requires without a context restart.